drivers/serial/uart_stm32.c

```diff
@@ -685,7 +690,8 @@ static const struct uart_stm32_config uart_stm32_cfg_##name = {		\
 };									\
 									\
 static struct uart_stm32_data uart_stm32_data_##name = {		\
-	.baud_rate = DT_UART_STM32_##name##_BAUD_RATE			\
+	.baud_rate = DT_UART_STM32_##name##_BAUD_RATE,			\
+	.hw_fc = DT_UART_STM32_##name##_HARDWARE_FC_RTSCTS \
```

Comment:
Is that field strictly required? We're able to store/load info from IP registers so ideally we don't need to use RAM bits to store the info.